docs/ROLLOUT_STRATEGY_GUIDE.md

```diff
@@ -0,0 +1,364 @@
+# Rollout Strategy Guide
+
+## Overview
+
+KServe supports configurable rollout strategies for raw deployments, allowing you to control how new versions of your models are deployed. Rollout strategies can be configured through the ConfigMap defaults or by setting Kubernetes `DeploymentStrategy` directly in the component extension spec.
+
+## Configuration Priority
+
+The rollout strategy is applied with the following precedence:
+
+1. **User-defined DeploymentStrategy** (highest priority) - directly specified in component extension spec
+2. **ConfigMap rollout strategy** (fallback) - applies only when `defaultDeploymentMode` is `"RawDeployment"`
+
+## ConfigMap Configuration
+
+When using ConfigMap configuration, you can specify `maxSurge` and `maxUnavailable` values directly. These values are applied to the Kubernetes deployment strategy when `defaultDeploymentMode` is set to `"RawDeployment"`.
+
+## Configuration
+
+### Method 1: Direct DeploymentStrategy Configuration (Recommended)
+
+You can configure Kubernetes deployment strategy directly at the component level:
+
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: deployment-strategy-example
+  namespace: default
+  annotations:
+    serving.kserve.io/deploymentMode: "RawDeployment"
```

Comment:
same here, let's change to `standard`. We also need to move this doc to the website https://github.com/kserve/website